### ISSUE TITLE
feat: verbosity level + reasoning

### ISSUE THREAD

**Issue by pkarw** [callstackincubator/fabrice-ai#110]
@grabbou as we discussed - it would be awesome to have a simple way to increase the verbosity level and, at some point log the `reasoning` along with the task names - so it would be easier to optimize and to understand the execution plans

**Comment by pkarw** [comment]
Plus - add info on which parameters the tools were called with.

**Comment by grabbou** [comment]
Yup, should be added as option to built-in logger, and I guess we can consider turning it on by default.

**Comment by grabbou** [comment]
Will create a new logger interface that is bound to provider. It will provide pre/post request metadata, regarding model used, prompt used, token count etc.
